File: 56-merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key = lambda x : x[0])
        cur = intervals[0]
        ans = []
        for i in range(1, len(intervals)):
            curEnd = cur[1]
            nextStart = intervals[i][0]
            nextEnd = intervals[i][1]
            if (curEnd >= nextStart):
                cur[1] = max(curEnd, nextEnd)
            else:
                ans.append(cur)
                cur = intervals[i]
        ans.append(cur)
        return ans

File: 56-merge-intervals/merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        if (intervals.length < 2){
            return intervals;
        }
        Arrays.sort(intervals, (a,b) -> Integer.compare(a[0], b[0]));
        int[] curr = intervals[0];
        ArrayList<int[]> answer = new ArrayList<>();
        for(int[] next : intervals){
            int currEnd = curr[1];
            int nextStart = next[0];
            int nextEnd = next[1];
            if(currEnd >= nextStart){
                curr[1] = Math.max(nextEnd, currEnd);
            }
            else{
                answer.add(curr);
                curr = next;
            }
        }
        answer.add(curr);
        return answer.toArray(new int[answer.size()][2]);
    }
}